### ISSUE TITLE
ERROR: zeitwerk requires Ruby version >= 2.4.4.

### ISSUE THREAD

**Issue by jetsetting-stressball** [fxn/zeitwerk#122]
While upgrading ruby and rails in support of a redmine install:

```
user@vm:~/redmine$ bundle update
Fetching gem metadata from https://rubygems.org/.............
Resolving dependencies...........
Bundler could not find compatible versions for gem "activemodel":
  In Gemfile:
    protected_attributes was resolved to 1.0.2, which depends on
      activemodel (< 5.0, >= 4.0.0.beta)
    rails (~> 5.2) was resolved to 5.2.1.rc1, which depends on
      activemodel (= 5.2.1.rc1)

user@vm:~/redmine$ sudo gem install activemodel
ERROR:  Error installing activemodel:
        zeitwerk requires Ruby version >= 2.4.4.

user@vm:~/redmine$ sudo gem install zeitwerk
ERROR:  Error installing zeitwerk:
        zeitwerk requires Ruby version >= 2.4.4.

user@vm:~/redmine$ ruby -v
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux]
```

Context: This is for a redmine install from the Google Cloud Marketplace, which launches with a horribly outdated setup (Ruby @ 2.3.3-p222, Rails @ 4.2.7.1).  I'm working through the ruby/rails upgrade and hit this snag.  As shown ruby is now at 2.7 (which, last I checked, is greater than 2.4.), and I'm trying to update rails.

Not sure what's stopped zeitwerk from being able to compare versions correctly, but any thoughts would be appreciated...

**Comment by fxn** [comment]
What does

```
sudo ruby -v
```

print?

Besides understanding that, let me say that Rails 5.2 does not have a `zeitwerk` mode, it was added in Rails 6.

**Comment by jetsetting-stressball** [comment]
> What does `sudo ruby -v` print?

Unfortunately, it prints the same as above:
```ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux]```

That said... after a few more hours of banging my head against stack of help sites, I found the problem -- there was another ruby install that the system was looking for.  I (as a ruby novice) upgraded ruby, not considering the fact that my method of doing so (via RVM) wasn't actually an upgrade... it just installed v2.7 in a _separate_ location.  On top of that, the references to ruby in Redmine ere hard-coded in a few different places.

The upshot was that while the "dynamic" parts of the system were in fact looking at the latest install (hence the updated output for `ruby -v`), when I went into my redmine project and updated the Gemfile (or the bundle) it dug a few layers into the redmine configs and came up with the old install location and version, and thought it was what should be used.

For anyone who finds this while looking for answers, [this github issue](https://github.com/rubygems/bundler/issues/4260) was super helpful.

As a side note... the *one* thing I did correctly was make a backup before starting down this little rabbit hole.  🤓  By the time I figured out what went wrong, the system was so borked that I just trashed it and swapped in the old machine snapshot.  We'll see how round two goes now that I know a bit more...

Thanks for the quick response, and sorry for reporting a bug that's not your fault!!  As far as I'm concerned, this can be closed.
